Makefile

```diff
@@ -131,56 +131,70 @@ bundle: generate ## Generate bundle manifests and metadata, then validate genera
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	operator-sdk bundle validate ./bundle
 
-install_bats: ## Installs the bats util via NPM
-	$(MAKE) -C e2e install_bats
-
-e2e_test: export E2E_IMAGE = $(E2E_IMG)
-e2e_test: install_bats $(SETUP_E2E_TEST) $(KIND_KUBECONFIG) docker-build ## Runs the e2e test suite
-	@$(KIND_BIN) load docker-image --name $(KIND_CLUSTER) $(E2E_IMG)
-	@docker rmi $(E2E_IMG)
-	$(MAKE) -C e2e run_bats -e KUBECONFIG=../$(KIND_KUBECONFIG)
-
-run_kind: export KUBECONFIG = $(KIND_KUBECONFIG)
-run_kind: export BACKUP_ENABLE_LEADER_ELECTION = $(ENABLE_LEADER_ELECTION)
-run_kind: $(SETUP_E2E_TEST) ## Runs the operator in kind
-	go run ./main.go
-
-.PHONY: setup_e2e_test
-setup_e2e_test: $(SETUP_E2E_TEST) ## Run the e2e setup
-
-.PHONY: clean_e2e_setup
-clean_e2e_setup: export KUBECONFIG = $(KIND_KUBECONFIG)
-clean_e2e_setup: ## Clean the e2e setup (e.g. to rerun the setup_e2e_test)
-	kubectl delete ns k8up-system --ignore-not-found --force --grace-period=0 || true
-	@$(KUSTOMIZE_BUILD_CRD) | kubectl delete -f - || true
-	@rm $(SETUP_E2E_TEST) || true
-
 clean: export KUBECONFIG = $(KIND_KUBECONFIG)
-clean: ## Cleans up the generated resources
-	$(KIND_BIN) delete cluster --name $(KIND_CLUSTER) || true
-	docker images --filter "reference=$(E2E_REPO)" --format "{{.Repository }}:{{ .Tag }}" | xargs --no-run-if-empty docker rmi || true
+clean: clean_e2e clean_kind ## Cleans up the generated resources
 	rm -r testbin/ dist/ bin/ cover.out $(BIN_FILENAME) || true
-	$(MAKE) -C e2e clean
 
 .PHONY: help
 help: ## Show this help
 	@grep -E -h '\s##\s' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+###
+### KIND
+###
+
+.PHONY: rund_kind
+run_kind: export KUBECONFIG = $(KIND_KUBECONFIG)
+run_kind: export BACKUP_ENABLE_LEADER_ELECTION = $(ENABLE_LEADER_ELECTION)
+run_kind: $(SETUP_KIND) ## Runs the operator in kind
+	go run ./main.go
```

Comment:
I propose to move the KIND related stuff into e2e/Makefile and include it in the main Makefile with `include e2e/Makefile`. This would make main Makefile easier to maintain while clearly separating between "building" and "running"